### ISSUE TITLE
Package works fine but it's throwing 5 warnings

### ISSUE THREAD

**Issue by Janneman84** [dmrschmidt/DSWaveformImage#19]
On my quest to keep my project free of warnings I noticed a few warnings this package is generating in TempiFFT.swift:

`Inout expression creates a temporary pointer, but argument 'imagp' should be a pointer that outlives the call to 'init(realp:imagp:)'` (line 112)

`Initialization of 'UnsafeMutablePointer<Float>' results in a dangling pointer` (line 166 and 226)

These seem to be a bit above my knowledge, but maybe I can figure it out. Could you take a look at this too?

**Comment by Janneman84** [comment]
Ok I figured it out:

Replace line 112 with : 

```
super.init()

real.withUnsafeMutableBufferPointer {realBP in
    imaginary.withUnsafeMutableBufferPointer {imaginaryBP in
        self.complexBuffer = DSPSplitComplex(realp: realBP.baseAddress!, imagp: imaginaryBP.baseAddress!)
    }
}
```

Replace (what used to be) line 166 with:

```
reals.withUnsafeMutableBufferPointer {realsBP in
    imags.withUnsafeMutableBufferPointer {imagsBP in
        self.complexBuffer = DSPSplitComplex(realp: realsBP.baseAddress!, imagp: imagsBP.baseAddress!)
    }
}
```

Replace (what used to be) line 226 with:

```
array.withUnsafeBufferPointer {arrayBP in
    vDSP_meanv(arrayBP.baseAddress! + startIdx, 1, &mean, UInt(stopIdx - startIdx))
}
```

Maybe you could incorporate this into a new version?

**Comment by dmrschmidt** [comment]
Thanks @Janneman84! I've added those changes and released a 6.1.2. Works via carthage.

Unfortunately I only have access to an M1 MacBook for the next few weeks where a specific ruby dependency is still not working, which means I cannot push the update to cocoapods for the time being.

I've been meaning to migrate over to SPM anyway for a long time now, so I think I'll just make 6.1.1 the last officially supported cocoapods version and release a 7.0.0 (essentially just 6.1.2 with SPM support and removed cocoapods support) at some point.

**Comment by dmrschmidt** [comment]
Just added SPM support.
